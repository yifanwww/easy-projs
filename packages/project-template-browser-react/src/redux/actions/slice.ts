import { omitUnderscorePrefixActions } from '@easy/utils-redux';
import { createSlice } from '@reduxjs/toolkit';

import { IStoreState } from '../types';
import { _finishPreparing, updateClientAreaSize } from './reducers.DisplayConfig';

export function getInitialState(): IStoreState {
    return {
        displayConfig: {
            clientAreaSize: { height: 720, width: 1280 },
            prepared: false,
        },
    };
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {
        _finishPreparing,
        updateClientAreaSize,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
