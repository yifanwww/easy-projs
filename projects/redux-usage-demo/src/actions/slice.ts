import { omitUnderscorePrefixActions } from '@easy-pkg/helpers-redux';
import { createSlice } from '@reduxjs/toolkit';

import type { StoreState } from '../types.js';

export function getInitialState(): StoreState {
    return {};
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {},
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
