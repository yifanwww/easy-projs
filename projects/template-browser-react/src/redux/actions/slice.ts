import { omitUnderscorePrefixActions } from '@easy-pkg/utils-redux';
import { createSlice } from '@reduxjs/toolkit';

import { StoreState } from '../types';
import { _finishPreparing } from './reducers';
import { updateClientAreaSize } from './reducers.DisplayConfig';

export function getInitialState(): StoreState {
    return {
        displayConfig: {
            clientAreaSize: { height: 720, width: 1280 },
        },
        prepared: false,
    };
}

const slice = createSlice({
    name: 'redux',
    initialState: getInitialState(),
    reducers: {
        _finishPreparing,
        updateClientAreaSize,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
