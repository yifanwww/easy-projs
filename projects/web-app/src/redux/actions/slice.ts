import { omitUnderscorePrefixActions } from '@easy-pkg/utils-redux';
import { createSlice } from '@reduxjs/toolkit';

import type { StoreState } from '../types';

export function getInitialState(): StoreState {
    return {};
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {},
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
