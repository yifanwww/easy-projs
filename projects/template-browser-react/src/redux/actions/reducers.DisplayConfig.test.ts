import { getInitialState, _actions, _reducer } from './slice';

{
    const name = 'updateClientAreaSize';

    describe(`Test redux reducer \`${name}\``, () => {
        it('should update client area size', () => {
            const prevState = getInitialState();

            expect(_reducer(prevState, _actions[name]({ height: 123, width: 456 })).displayConfig).toMatchSnapshot();
        });
    });
}
