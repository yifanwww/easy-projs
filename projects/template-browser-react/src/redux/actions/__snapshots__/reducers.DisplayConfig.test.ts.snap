// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test redux reducer \`updateClientAreaSize\` should update client area size 1`] = `
{
  "clientAreaSize": {
    "height": 123,
    "width": 456,
  },
}
`;
